src/app/home/services/home_service.js

```diff
@@ -2,28 +2,58 @@ var posts = [
   {
     id: 1,
     title: 'Save energy',
-    category: 'energy conservation',
+    category_id: 1,
     content: 'Turn off lights when I leave a room and use natural lighting as much as possible'
   }, {
     id: 2,
     title: 'Say no to plastics',
-    category: 'eco-friendly habits',
+    category_id: 2,
     content: 'Use less and compost and recycle as much as possible'
   }, {
     id: 3,
     title: 'Use Automobiles smartly',
-    category: 'Pollution',
-    content: 'Walk and ride my bike less to and from school and get a ride with friends instea' +
-        'd'
+    category_id: 3,
+    content: 'Walk and ride my bike less to and from school and get a ride with friends instead'
   }, {
     id: 4,
     title: 'Go green',
-    category: 'eco-friendly habits',
-    content: 'Work with my People.Power.Planet Champions to find ways that we can help our sch' +
-        'ool save energy and the environment'
+    category_id: 4,
+    content: 'Work with my People.Power.Planet Champions to find ways that we can help our school save energy and the environment'
+  }
+];
+
+var categories = [
+  {
+    id:1,
+    category:'energy conservation'
+  }, {
+    id:2,
+    category:'no plastics'
+  }, {
+    id:3,
+    category:'Pollution'
+  }, {
+    id:4,
+    category:'eco-friendly habits'
   }
 ];
 
 export function getPosts() {
   return posts;
-}
\ No newline at end of file
+
+}
+
+export function getPostById(id) {
+  return posts[id-1];
```

Comment:
The post with id 1 or n is 0th or (n-1)th element in posts array and hence this logic.